### PR TITLE
Remove sharedPreferences and NSUserDefaults for storing the app id

### PR DESCRIPTION
**Pull request by ChristiaanScheermeijer**
This PR updates the app ID configuration because we found some issues due to using NSUserDefaults and SharedPreferences.

Tested on iOS and Android devices

### CODE REVIEW

**Review by royschut — APPROVED** [review]
With "some issues" you refer to unwanted appId 'caching', correct? I'm curious why this mechanism exists, to be honest.

**Comment by ChristiaanScheermeijer** [comment]
Yes indeed. Because of the usage of these storages, the receiver id is persisted once connected. This never seems to be overridden, and it creates an issue when the receiver ID is changed.

This mechanism is in place because the app needs the receiver ID on startup. But you shouldn't manually edit the iOS/Android projects in Cordova apps. For Capacitor apps, this isn't the case anymore. So now we can configure the receiver IDs in the native configuration instead of using the workaround previously implemented.

**Review by langemike — APPROVED** [review]
Very nice!

File: www/chrome.cast.js [hunk 1]
```diff
@@ -544,7 +544,7 @@ var _session;
  * @param  {function} errorCallback
  */
 chrome.cast.initialize = function (apiConfig, successCallback, errorCallback) {
-    execute('initialize', apiConfig.sessionRequest.appId, apiConfig.autoJoinPolicy, apiConfig.defaultActionPolicy, function (err) {
+    execute('initialize', apiConfig.autoJoinPolicy, apiConfig.defaultActionPolicy, function (err) {
```
Comment (MelissaDTH) [inline www/chrome.cast.js:551]: What is the reason the `apiConfig.sessionRequest.appId` can be removed here as the first argument? If I read the documentation, it states that it's value cannot be null, but is it safe to say we do not need te define it anymore in this 'initialize' method? Is it derived from somewhere else in the project? See: https://developers.google.com/cast/docs/reference/web_sender/chrome.cast.ApiConfig#ApiConfig

**Review by MelissaDTH — APPROVED** [review]
One question remains, but still approved

File: www/chrome.cast.js (hunk 1 above)
Comment (ChristiaanScheermeijer) [reply www/chrome.cast.js:551]: Good question! The Cordova Cast Plugin attempts to mimic the cast sender web SDK API. The `chrome.cast.SessionRequest` instance expects an `appId`, but we no longer use it in the native code.

This way there is no breaking change in the plugin API and it still aligns with the web sender SDK API 😄

**Comment by wouterbin** [comment]
Hi,

We are using your fork of this plugin and implemented it recently. We had to replace https://github.com/ConnectSDK/Connect-SDK-Cordova-Plugin with a Capacitor compatible alternative. This plugin is, thanks!

The proposed changes seem counter intuitive to me. The plugin gives the possibility to start a session given a certain appId:

```new chrome.cast.SessionRequest('ABCD1234')```

With these changes the argument can be given but is silently ignored. I think the user should be warned that this is not the way to use this method or it should work as expected. As [mentioned](https://github.com/Videodock/cordova-plugin-chromecast/pull/2/files#r1351857571) the plugin advocates that it implements the cast sender web SDK API. In my opion this breaks it ☺️ .

Question: What is the use case for changing the appId? Is this used in production apps?

**Comment by ChristiaanScheermeijer** [comment]
Hi @wouterbin,

Thank you very much for your feedback! I'm happily surprised that this fork is being used 😄 

You're right. This change will make it impossible to change the appId from the web app. But there are a few reasons for this.

- As mentioned above, the `appId` is persisted and never overridden. We have an app in production with multiple environments test, QA, and production, which all use different application IDs. While developing and testing, we had really weird situations where the QA and prod receiver apps were mixed.
- The receiver id is hardcoded in the Info.plist for iOS. This means that theoretically, it would be possible to change the appId from the web app dynamically, but it only works if the application ID aligns with the application ID defined in the `Bonjour Services`

I will add a warning and update the readme.

**Comment by wouterbin** [comment]
Thanks for the changes ✅ !

> I'm happily surprised that this fork is being used 😄

Yeah, it was at the time (probably still is) the most maintained fork. So it was an obvious choice 😉. Keep up the good work 👍 . Would you be interested if we stumble upon issues with the plugin (The issues on this repo are disabled now)?

**Comment by ChristiaanScheermeijer** [comment]
@wouterbin yes, sure! I've enabled issues. Please let us know if you have any issues.

**Pull request merged**